training/driverless_ai/DriverlessAI_Transformations.md

```diff
@@ -0,0 +1,130 @@
+
+* get hour in day, get minute in hour, get day in month, get day in year, get quarter in year, get month in year, get week in year, get week day in week
+
+| Date Built | Square Footage | Num Beds | Num Baths | State | Price | DateBuilt\_MonthInYear\_x| DateBuilt\_MonthInYear\_y|
+|---|---|---|---|---|---|---|---|
+| 01/01/1920 | 1700 | 3 | 2 | NY | 700,000 | 0.5 | 1 |
+
+The polar coordinates of the month January in year is (0.5, 1).  This allows the model to catch the similarities between January and December.  This information was not captured in the simple Date Transformer.
+
+
+### Text Transformer
+
+* transform text column using methods: TFIDF or count (count of the word)
+* this may be followed by dimensionality reduction using truncated SVD
+
+### Categorical Target Encoding Transformer
+
+* cross validation target encoding done on a categorical column
+
+| Date Built | Square Footage | Num Beds | Num Baths | State | Price | CV\_TE\_State|
+|---|---|---|---|---|---|---|
+| 01/01/1920 | 1700 |3 | 2 | NY | 700,000 | 550,000 | 
+
+The average price of properties in NY state is $550,000*.  
+
+*In order to prevent overfitting, Driverless AI calculates this average on out-of-fold data using cross validation.
+
+### Numeric to Categorical Target Encoding Transformer
+
+* numeric column converted to categorical by binning
+* cross validation target encoding done on the binned numeric column
+
+| Date Built | Square Footage | Num Beds | Num Baths | State | Price | CV\_TE\_SquareFootage|
+|---|---|---|---|---|---|---|
+| 01/01/1920 | 1700 | 3 | 2 | NY | 700,000 | 345,000 | 
+
+The column `Square Footage` has been bucketed into 10 equally populated bins. This property lies in the `Square Footage` bucket 1,572 to 1,749.  The average price of properties with this range of square footage is $345,000*.
+
+*In order to prevent overfitting, Driverless AI calculates this average on out-of-fold data using cross validation.
+
+
+### Cluster Target Encoding Transformer
+
+* selected columns in the data are clustered
+* target encoding is done on the cluster ID
+
+| Date Built | Square Footage | Num Beds | Num Baths | State | Price | ClusterTE\_4\_NumBeds\_NumBaths\_SquareFootage|
+|---|---|---|---|---|---|---|
+| 01/01/1920 | 1700 | 3 | 2 | NY | 700,000 | 450,000 | 
+
+The columns: `Num Beds`, `Num Baths`, `Square Footage` have been segmented into 4 clusters.  The average price of properties in the same cluster as the selected property is $450,000*.
+
+*In order to prevent overfitting, Driverless AI calculates this average on out-of-fold data using cross validation.
+
+
+### Cluster Distance Transformer
+
+* selected columns in the data clustered
```

Comment:
Minor change: selected columns in the data clustered > selected columns in the data are clustered.
(This is only to make it consistent with the Cluster Target Encoding Transformer section.)